lib/src/features/home_screen/presentation/pages/articles_page.dart

```diff
@@ -42,23 +45,134 @@ class _AritclesPageState extends State<AritclesPage> with TextStyles {
             return Text('Retry again ${state.message}');
           }
           if (articleModelList.isEmpty) {
             return const Text('No Articles');
           } else {
-            return Text(
-              'Articles ${articleModelList.length}',
-              style: smallTextStyle(),
+            return SingleChildScrollView(
+              child: Padding(
+                padding: const EdgeInsets.symmetric(horizontal: 15),
+                child: ListView.separated(
+                  shrinkWrap: true,
+                  padding: EdgeInsets.zero,
+                  physics: const NeverScrollableScrollPhysics(),
+                  itemCount: articleModelList.length,
+                  itemBuilder: (BuildContext context, int index) {
+                    Article articleData = articleModelList[index];
+
+                    String imageUrl = articleData.media.isNotEmpty &&
+                            articleData.media.first.mediaMetadata.isNotEmpty
+                        ? articleData.media.first.mediaMetadata.first.url
+                        : AppStrings.noImageURL;
+
+                    return ConstrainedBox(
+                      constraints: BoxConstraints(
+                        maxHeight: MediaQuery.of(context).size.height * 0.15,
+                        maxWidth: MediaQuery.of(context).size.width,
+                      ),
+                      child: Row(
+                        children: <Widget>[
+                          CachedNetworkImage(
+                            imageUrl: imageUrl,
+                            width: 130,
+                            height: 150,
+                            fit: BoxFit.fitWidth,
+                          ),
+                          const SizedBox(width: 10),
+                          Expanded(
+                            child: Column(
+                              crossAxisAlignment: CrossAxisAlignment.start,
+                              children: <Widget>[
+                                Text(
+                                  articleData.title,
+                                  overflow: TextOverflow.ellipsis,
+                                ),
+                                Text(
+                                  articleData.abstract,
+                                  style: const TextStyle(color: Colors.grey),
+                                  maxLines: 2,
+                                  overflow: TextOverflow.ellipsis,
+                                ),
+                              ],
+                            ),
+                          ),
+                        ],
+                      ),
+                    );
+                  },
+                  separatorBuilder: (BuildContext context, int index) =>
+                      const SizedBox(height: 10),
+                ),
+              ),
             );
           }
         },
       ),
     );
   }
 
+  Widget _buildShimmerEffect() {
```

Comment:
please make this as common widget

Reply:
it's dedicated to list of articles only, hence moving it to widget folders